frontend/src/app/modules/shared/queue/queue/queue.component.ts

```diff
@@ -63,9 +65,31 @@ export class QueueComponent {
 
   clickMenuHandler(data: { menuItem: string, song: Song }) {
     switch (data.menuItem) {
-      case 'Add to queue':
-        this.songs = [...this.songs, data.song];
+      case 'Add to queue': {
+        if (!this.songs.find((s) => s.id === data.song.id)) {
+          this.songs = [...this.songs, data.song];
+        }
+        break;
+      }
+      case 'Remove from queue': {
+        const removingSong = this.songs.find((s) => s.id === data.song.id);
+        if (removingSong) {
+          if (this.songs.length === 1) {
+            break;
+          }
+          if (this.currentSongId === removingSong.id) {
+            if (this.getCurrentSongIndex() + 1 === this.songs.length) {
+              this._queueService.previousSong();
+            }
+            else {
+              this._queueService.nextSong();
+            }
```

Comment:
We could transter it to separate method like getNextSong(). You could play not previous but first in queue